doc/user: Add metrics and service-monitor docs (#719)
operator-framework/operator-sdk: doc/user/metrics/service-monitor.md

The `GenerateServiceMonitor` takes a `Service` object and generates a `ServiceMonitor` resource based on it. To add `Service` target discovery of your created monitoring `Service`s you can use the `metrics.CreateServiceMonitor()` helper function, which accepts the newly created `Service`.

### Prerequisites:

- [prometheus-operator][prom-quickstart] needs to be deployed already in the cluster.

### Usage example:

```go
    import(
        v1 "k8s.io/api/core/v1"
        "github.com/operator-framework/operator-sdk/pkg/metrics"
    )
    
    // Populate bellow with Service(s) for which you want to create `ServiceMonitor` for.
    services := []*v1.Service{}

    // Create one `ServiceMonitor` per application per namespace.
    // Change below value to name of the Namespace you want the `ServiceMonitor` to be created in.
    ns := "default"
    
    // Pass the Service(s) to the helper function, which in turn returns the `ServiceMonitor` object.
    serviceMonitors, err := metrics.CreateServiceMonitors(restConfig, ns, services)
    if err != nil {
        // handle error here

>     }

Reviewer: Would the above also typically be called in `func main()`? If, so I'd suggest also wrapping this in `func main()` opening and closing.

> ## Using the ServiceMonitor prometheus-operator CRD

Reviewer 2: Nit: Rephrasing the title for clarity.
```suggestion
## Using the ServiceMonitor to access metrics with Prometheus
```

Author: Not sure I agree with this, as ServiceMonitor is only a concept within prometheus-operator, I would include the name of it in the title.

Reviewer 2: Fair point. I think I've gotten used to associating it directly with Prometheus since the operator is always involved.

> [prometheus-operator][prom-operator] an operator that creates, configures and manages Prometheus clusters atop Kubernetes.

Reviewer 3: ```suggestion
[prometheus-operator][prom-operator] is an operator that creates, configures, and manages Prometheus clusters atop Kubernetes.
```

> `ServiceMonitor` is a [CR][cr] of the prometheus-operator, which discovers the `Endpoints` in `Service` objects and configures Prometheus to monitor those `Pod`s. See the prometheus-operator [documention][sm] to learn more about `ServiceMonitor`s.

Reviewer 2: Typo. Also change the link tag from sm to service-monitor.
Mostly for our sake but at a glance it's more clear what we're linking to.
```suggestion
`ServiceMonitor` is a [CR][cr] of the prometheus-operator, which discovers the `Endpoints` in `Service` objects and configures Prometheus to monitor those `Pod`s. See the prometheus-operator [documentation][service-monitor] to learn more about `ServiceMonitor`s.
```

> `ServiceMonitor` is a [CR][cr] of the prometheus-operator, which discovers the `Endpoints` in `Service` objects and configures Prometheus to monitor those `Pod`s. See the prometheus-operator [documention][service-monitor] to learn more about `ServiceMonitor`s.

Reviewer 2: CR==>CustomResource.
I think it's better to use the full name if we haven't defined an acronym before using it in a given doc.
```suggestion
`ServiceMonitor` is a CustomResource of the prometheus-operator, which discovers the `Endpoints` in `Service` objects and configures Prometheus to monitor those `Pod`s. See the prometheus-operator [documention][service-monitor] to learn more about `ServiceMonitor`s.
```
Also just a suggestion but we probably can do without linking to the CustomResource docs here. At this point users should be familiar with CR and CRDs.
Linking to the ServiceMonitor doc is all we need which we've already done.

Reviewer 2: Maybe this is subjective but "`Pod`s" looks a bit odd to me.
Can we change it to either "`Pods`" or just "Pods"(without the highlighting)?
Same for "`Service`s" and "`ServiceMonitor`s"

> The `GenerateServiceMonitor` takes a `Service` object and generates a `ServiceMonitor` resource based on it. To add `Service` target discovery of your created monitoring `Service`s you can use the `metrics.CreateServiceMonitor()` helper function, which accepts the newly created `Service`.

Reviewer 3: ```suggestion
The `GenerateServiceMonitor` function takes a `Service` object and generates a `ServiceMonitor` resource based on it. To add `Service` target discovery of your created monitoring `Service`s you can use the `metrics.CreateServiceMonitor()` helper function, which accepts the newly created `Service`.
```

### Prerequisites:

> - [prometheus-operator][prom-quickstart] needs to be deployed already in the cluster.

Reviewer 3: ```suggestion
- [prometheus-operator][prom-quickstart] needs to be deployed in the cluster.
```

### Usage example:

```go
    import(

>         v1 "k8s.io/api/core/v1"

Reviewer 2: Is the alias `v1` needed if the package name is already `v1`? 
```suggestion
        "k8s.io/api/core/v1"
```

        "github.com/operator-framework/operator-sdk/pkg/metrics"
    )

    func main() {    
        
        ...

>         // Populate bellow with Service(s) for which you want to create `ServiceMonitor` for.

Reviewer 2: Typo: *below
```suggestion
        // Populate below with the Service(s) for which you want to create ServiceMonitors.
```
Also no need to put the backticks for  \`ServiceMonitor \` since it's not being highlighted for markdown I think.

        services := []*v1.Service{}

        // Create one `ServiceMonitor` per application per namespace.
        // Change below value to name of the Namespace you want the `ServiceMonitor` to be created in.
        ns := "default"
        

>         // Pass the Service(s) to the helper function, which in turn returns the `ServiceMonitor` object.

Reviewer 2: It returns an array of ServiceMonitors right?
https://github.com/operator-framework/operator-sdk/blob/master/pkg/metrics/service-monitor.go#L29 
```suggestion
        // Pass the Service(s) to the helper function, which in turn returns the array of `ServiceMonitor` objects.
```
